### PR TITLE
[DOCS] Reformat keep types and keep words token filter docs

### PR DESCRIPTION
**Pull request by jrodewig**
Reformats the keep types and keep words token filter docs as part of #44726.

### Changes
* Adds title abbreviations
* Updates the descriptions and adds Lucene links
* Reformats parameter definitions
* Adds analyze and custom analyzer snippets
* Adds explanations of token types to keep types token filter and tokenizer docs

### CODE REVIEW

File: docs/reference/analysis/tokenfilters/keep-types-tokenfilter.asciidoc [hunk 1]
```diff
@@ -1,137 +1,198 @@
...
-contained in a predefined set.
+Keeps or removes tokens of a specific type. For example, you can use this filter
+to change `3 quick foxes` to `quick foxes` by keeping only `<ALPHANUM>`
+(alphanumeric) tokens.
 
+[NOTE]
+.Token types
+====
+Token types are set by the <<analysis-tokenizers,tokenizer>> when converting
+characters to tokens. Token types can vary between tokenizers.
```
Comment (cbuescher) [inline docs/reference/analysis/tokenfilters/keep-types-tokenfilter.asciidoc]: Maybe worth an addition that also certain filters (e.g. synonym filter) can add tokens with a specific type (in that case of `SYNONYM` type)

**Review by cbuescher — APPROVED** [review]
I left a suggestion for an addition, the rest LGTM

File: docs/reference/analysis/tokenfilters/keep-types-tokenfilter.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/analysis/tokenfilters/keep-types-tokenfilter.asciidoc]: Thanks! Great point. Added a note with 5082846eeca.

**Comment by jrodewig** [comment]
master: 6ea54ee 
7.x: ade72b97b76e731ca115c5079cd7dfa9d67c1cd0
7.5: 7ada1e6087c74dd4a1ea7826a3d612fd304969ab
7.4: 22f5835b171fdd5eab47a18a030cc14f53855e98

**Pull request merged**